core/lib/libmbedtls/src/tee_lmd_provider.c

```diff
@@ -0,0 +1,606 @@
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+#endif /* CFG_CRYPTO_ECC */
+
+#endif /* _CFG_CRYPTO_WITH_ACIPHER */
+
+/******************************************************************************
+ * Symmetric ciphers
+ ******************************************************************************/
+
+#if defined(_CFG_CRYPTO_WITH_CIPHER)
+TEE_Result crypto_cipher_alloc_ctx(void **ctx __unused, uint32_t algo __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+void crypto_cipher_free_ctx(void *ctx, uint32_t algo __unused)
+{
+	if (ctx)
+		assert(0);
+}
+
+void crypto_cipher_copy_state(void *dst_ctx __unused, void *src_ctx __unused,
+			      uint32_t algo __unused)
+{
+}
+
+TEE_Result crypto_cipher_init(void *ctx __unused, uint32_t algo __unused,
+			      TEE_OperationMode mode __unused,
+			      const uint8_t *key1 __unused,
+			      size_t key1_len __unused,
+			      const uint8_t *key2 __unused,
+			      size_t key2_len __unused,
+			      const uint8_t *iv __unused,
+			      size_t iv_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_cipher_update(void *ctx __unused, uint32_t algo __unused,
+				TEE_OperationMode mode __unused,
+				bool last_block __unused,
+				const uint8_t *data __unused,
+				size_t len __unused, uint8_t *dst __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+void crypto_cipher_final(void *ctx __unused, uint32_t algo __unused)
+{
+}
+
+TEE_Result crypto_cipher_get_block_size(uint32_t algo __unused,
+					size_t *size __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+#endif /*_CFG_CRYPTO_WITH_CIPHER*/
+
+/*****************************************************************************
+ * Message Authentication Code functions
+ *****************************************************************************/
+
+#if defined(_CFG_CRYPTO_WITH_MAC)
+TEE_Result crypto_mac_alloc_ctx(void **ctx __unused, uint32_t algo __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+void crypto_mac_free_ctx(void *ctx, uint32_t algo __unused)
+{
+	if (ctx)
+		assert(0);
+}
+
+void crypto_mac_copy_state(void *dst_ctx __unused, void *src_ctx __unused,
+			   uint32_t algo __unused)
+{
+}
+
+TEE_Result crypto_mac_init(void *ctx __unused, uint32_t algo __unused,
+			   const uint8_t *key __unused, size_t len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_mac_update(void *ctx __unused, uint32_t algo __unused,
+			     const uint8_t *data __unused, size_t len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+
+TEE_Result crypto_mac_final(void *ctx __unused, uint32_t algo __unused,
+			    uint8_t *digest __unused,
+			    size_t digest_len __unused)
+{
+	return TEE_ERROR_NOT_IMPLEMENTED;
+}
+#endif /*_CFG_CRYPTO_WITH_MAC*/
+
+/******************************************************************************
+ * Authenticated encryption
+ ******************************************************************************/
+
+#define TEE_CCM_KEY_MAX_LENGTH		32
```

Comment:
These defines are unused for now, please add them when needed instead.

Reply:
Yes